exla/test/exla/nx_linalg_doctest_test.exs

```diff
@@ -91,4 +88,318 @@ defmodule EXLA.MLIR.NxLinAlgDoctestTest do
       end
     end
   end
+
+  describe "cholesky" do
```

Comment:
The tests added here are generally extracted from the Nx linalg tests in nx itself. @TomasPegado is working on fixing triangular_solve and then we should also extract triangular_solve and solve tests to here as well.